### DISCUSSION TITLE
Annotated[None | IdSubcategoriesEnum | JddSubcategoriesEnum] does not work in Swagger UI

### DISCUSSION THREAD

**Discussion by fareshan** [fastapi/fastapi#12410] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post("/proof/categorize/file_binary/")
async def categorize_proof_file_api(
    file: Annotated[UploadFile, File()],
    category: Annotated[CategoriesEnum, Form()],
    subcategory: Annotated[
        None | IdSubcategoriesEnum | JddSubcategoriesEnum, Form()
    ] = None,
):
```

### Description

Expected behavior in Swagger ui:
- dropdown including IdSubcategoriesEnum values, JddSubcategoriesEnum values and None
Actual behavior:
- dropdown including only IdSubcategoriesEnum values and None

In Redoc, I get the expected documentation: `IdSubcategoriesEnum (string) or JddSubcategoriesEnum (string) or Subcategory (null) (Subcategory)`

-> **Since the behavior is correct in Redoc, it can be a Swagger issue. But how can I check this ?** so that I can create an issue in swagger if it is the case

### Operating System

macOS

### Operating System Details

Apple M1
Macos 15.0.1 (24A348)

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

3.12.6

**Comment by JavierSanchezCastro** [comment]
Try with `SkipJsonSchema` from Pydantic

**Comment by fareshan** [comment]
@JavierSanchezCastro 
```
Annotated[SkipJsonSchema[None] | SkipJsonSchema[IdSubcategoriesEnum] | SkipJsonSchema[JddSubcategoriesEnum], Form()]
```
replaces the dropdown by a text input. It is better but not perfect.

---
```
Annotated[SkipJsonSchema[None | IdSubcategoriesEnum | JddSubcategoriesEnum], Form()]
```
removes the dropdown completely

**Reply by JavierSanchezCastro** [reply]
Sorry I misunderstood your problem. Regarding the `openapi.json` schema seems good when you have something like `None | Enum1 | Enum2` generating that part of the schema like:

```
"parameters": [
          {
            "name": "c",
            "in": "query",
            "required": false,
            "schema": {
              "anyOf": [
                {
                  "$ref": "#/components/schemas/Countries"
                },
                {
                  "$ref": "#/components/schemas/MoreCountries"
                }
              ],
              "title": "C"
            }
          }
        ],
```

So I assume it's Swagger problem.

**Reply by JavierSanchezCastro** [reply]
Oh no, it's not correct the "anyOf" there, must be "allOf". Changing that, Swagger shows correctly both enums in same dropdown. Then, FastAPI issue. You can write a workaround easily joining both enums and then request that final enum in your function.

**Reply by fareshan** [reply]
Using this code
```
def combine_enums(*enums):
    combined = {e.name: e.value for enum in enums for e in enum}
    return Enum('CombinedEnum', combined)

# Combine Enum1 and Enum2
CombinedEnum = combine_enums(Enum1, Enum2)
```
gives me the expected dropdown with the expected values.

But now these values are handled as CombinedEnum instead of Enum1 / Enum2. So I would need to write some code to check for that.

So, this is helpful. But in my context, I prefer `Annotated[SkipJsonSchema[None | Enum1 | Enum2], Form()]` : text field instead of the dropdown, and the code stays clean

**Reply by YuriiMotov** [reply]
> Oh no, it's not correct the "anyOf" there, must be "allOf".

I still think `"anyOf"` is correct.
https://swagger.io/docs/specification/v3_0/data-models/oneof-anyof-allof-not/#anyof

For me it seems like an issue of Swagger UI

**Comment by fareshan** [comment]
Can we consider this as a issue ?
